aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -341,6 +342,23 @@ public Builder syncExpression(@NonNull Class<? extends Model> modelClass,
             return Builder.this;
         }
 
+        /**
+         * Sets a sync expression for a particular model to filter which data is synced locally.  The expression
+         * is evaluated each time DataStore is started.  The QueryPredicate is applied on both sync and subscriptions.
+         * @param modelSchema the schema of the model class for which the filter applies
+         * @param syncExpression DataStoreSyncExpression that should be used to filter the data that is synced.
+         * @return Current builder
+         */
+        @NonNull
+        public Builder syncExpression(@NonNull ModelSchema modelSchema,
+                                      @NonNull DataStoreSyncExpression syncExpression) {
+            this.syncExpressions.put(
+                    Objects.requireNonNull(modelSchema).getName(),
+                    Objects.requireNonNull(syncExpression)
+            );
+            return Builder.this;
```

Comment:
if we are only storing the name of model anyways, then why not take in the model name as parameter?

Reply:
^also this way, `Builder#syncExpression(Model, DataStoreSyncExpression)` can be implemented as
```
return syncExpression(Objects.requireNonNull(model).getSimpleName(), syncExpression);
```
which would _kind of_ reduce duplicate-ish code

Reply:
Fwiw, the `query` call from Flutter also just takes `modelName` as the param directly. So there may be an argument to be made that it's more consistent to do that here as well?

Reply:
I considered this as well.  The main reason I went with `ModelSchema` was for parity with iOS (see [iOS sync expression docs](https://docs.amplify.aws/lib/datastore/sync/q/platform/ios#selectively-syncing-a-subset-of-your-data)) . 

Thinking about it further though, it's still not actually in parity with iOS, since iOS takes an array of `SyncExpression` objects, which wrap a `ModelSchema` and a `QueryPredicate`.

I'll go ahead and switch to use a `String` instead!